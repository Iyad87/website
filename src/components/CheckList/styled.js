import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import breakpoints from '../../styles/breakpoints';
import { spacing } from '../../styles/spacing';
import { textLight } from '../../styles/colours';

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  margin: ${spacing(6)} 0;
  padding: 0;
`;

export const ListItem = styled.li`
  color: ${textLight};
  display: flex;
  flex-direction: row;
  font-size: 1.25rem;
  margin: ${spacing(2)} 0;
  width: 100%;

  ${breakpoints.tablet`
    padding-right: ${spacing(2)};
    width: 50%;

    &:nth-child(2n) {
      padding-left: ${spacing(2)};
      padding-right: 0;
    }
  `};
`;

export const Icon = styled(FontAwesomeIcon)`
  color: #37beb0;
  font-size: 2rem;
  margin-right: ${spacing(3)};
`;
