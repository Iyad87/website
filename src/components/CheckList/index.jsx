import React from 'react';
import PropTypes from 'prop-types';
import { faCheckCircle } from '@fortawesome/free-solid-svg-icons';
import { List, ListItem, Icon } from './styled';

const Item = ({ content }) => (
  <ListItem>
    <Icon icon={faCheckCircle} /> {content}
  </ListItem>
);

Item.propTypes = {
  content: PropTypes.node.isRequired,
};

const CheckList = () => (
  <List>
    <Item content="Build custom software solutions that become an asset" />
    <Item content="Deliver the right user experience at the right time, on the right channel" />
    <Item content="Transform, innovate and differentiate at speed" />
    <Item content="Grow capability inside your organisation - consulting, training, pairing and education" />
    <Item content="Adopt cutting edge technologies - We're pioneers in Node.js, React and Design Systems" />
    <Item content="Grow a technology culture" />
  </List>
);

export default CheckList;
