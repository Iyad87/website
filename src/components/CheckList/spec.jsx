import React from 'react';
import { shallow } from 'enzyme';
import CheckList from './index';

test('components/CheckList snapshot', () => {
  const wrapper = shallow(<CheckList />);

  expect(wrapper).toMatchSnapshot();
});
