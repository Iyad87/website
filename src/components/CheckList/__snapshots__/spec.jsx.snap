// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CheckList snapshot 1`] = `
<styled__List>
  <Item>
    Hello, world
  </Item>
  <Item>
    I am another item
  </Item>
  <Item>
    And a third
  </Item>
</styled__List>
`;
